### PR TITLE
use std::numeric_limits: update  INT_MIN/MAX, FLT_MIN/MAX, DBL_MIN/MAX etc

### PR DESCRIPTION
**Pull request by lx18233184051**

### CODE REVIEW

File: cyber/event/perf_event.h [hunk 1]
```diff
@@ -157,7 +158,7 @@ class TransportEvent : public EventBase {
  private:
   std::string adder_ = "";
   uint64_t msg_seq_ = 0;
-  uint64_t channel_id_ = UINT64_MAX;
+  uint64_t channel_id_ = std::numeric_limits<u_int64_t>::max();
```
Comment (storypku) [inline cyber/event/perf_event.h:161]: Better use the more wildly accepted `uint64_t` rather than `u_int64_t` defined in <sys/types.h>.

Comment (xiaoxq) [reply cyber/event/perf_event.h:161]: Agree. And at least the two `T` in `T channel_id_ = std::numeric_limits<T>::max();` should be aligned.

File: cyber/scheduler/policy/choreography_context.cc [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include "cyber/scheduler/policy/choreography_context.h"
 
 #include <climits>
```
Comment (storypku) [inline cyber/scheduler/policy/choreography_context.cc:19]: The `<climits>` header can be safely deleted now as XXX_MAX/MIN was no longer used ?

File: cyber/record/record_viewer.h [hunk 3]
```diff
@@ -46,9 +47,10 @@ class RecordViewer {
    * @param end_time
    * @param channels
    */
-  RecordViewer(const RecordReaderPtr& reader, uint64_t begin_time = 0,
-               uint64_t end_time = UINT64_MAX,
-               const std::set<std::string>& channels = std::set<std::string>());
+  RecordViewer(
+      const RecordReaderPtr& reader, uint64_t begin_time = 0,
+      uint64_t end_time = std::numeric_limits<uint64_t>::max(),
+      const std::set<std::string>& channels = std::set<std::string>());
```
Comment (storypku) [inline cyber/record/record_viewer.h:53]: Just write `{}` rather than `std::set<std::string>()`.

File: cyber/scheduler/policy/classic_context.cc [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #include "cyber/scheduler/policy/classic_context.h"
 
 #include <climits>
```
Comment (storypku) [inline cyber/scheduler/policy/classic_context.cc:19]: ditto.

File: modules/data/tools/smart_recorder/realtime_record_processor.cc [hunk 5]
```diff
@@ -19,6 +19,7 @@
 #include <algorithm>
 #include <chrono>
 #include <csignal>
```
Comment (storypku) [inline modules/data/tools/smart_recorder/realtime_record_processor.cc:21]: csignal should be put before algorithm and has its own section.

File: modules/perception/camera/common/twod_threed_util.h [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 #include <algorithm>
 #include <cfloat>
```
Comment (storypku) [inline modules/perception/camera/common/twod_threed_util.h:19]: `<cfloat>` can be removed ?

File: modules/perception/camera/lib/lane/common/common_functions.h [hunk 7]
```diff
@@ -18,6 +18,7 @@
 #include <vector>
 
 #include "Eigen/Core"
+#include <limits>
```
Comment (storypku) [inline modules/perception/camera/lib/lane/common/common_functions.h:21]: system header `<limits>` should be put before third_party `Eigen`

File: modules/data/tools/smart_recorder/realtime_record_processor.cc (hunk 5 above)
Comment (lx18233184051) [reply modules/data/tools/smart_recorder/realtime_record_processor.cc:21]: csignal is a c++ header, it's translated from c,  so separately place first?

Comment (storypku) [reply modules/data/tools/smart_recorder/realtime_record_processor.cc]: > csignal is a c++ header, it's translated from c, so separately place first?

yep.

**Pull request merged**